providers/gce/gce_instances.go

```diff
@@ -856,3 +856,21 @@ func (g *Cloud) GetNodeTags(nodeNames []string) ([]string, error) {
 	g.lastComputedNodeTags = tags
 	return tags, nil
 }
+
+// NodeNetworkInterfacesByProviderID returns a list of node interfaces that exist on the node.
+func (g *Cloud) NodeNetworkInterfacesByProviderID(providerID string) (interfaces []*compute.NetworkInterface, err error) {
```

Comment:
Why is this required? Can we just use the existing functions to fetch the instance and then iterate over the interfaces?

Reply:
Went through the existing functions already and found no function actually returns a *compute.Instance in the result. There are a few functions that return a *gceInstance, but *gceInstance is a subset of the *compute.Instance without the network interfaces information. Also, *gceInstance is not an exported variable.